packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts

```diff
@@ -103,6 +116,29 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
   async getScheduledTasks(): Promise<TaskDescriptor[]> {
     return this.allScheduledTasks;
   }
+
+  private wrapInMetrics(
+    fn: TaskFunction,
+    opts: { labels: Record<string, string> },
+  ): TaskFunction {
+    return async abort => {
+      this.counter.add(1, { ...opts.labels, result: 'started' });
+
+      const startTime = process.hrtime();
+
+      try {
+        await fn(abort);
+        this.counter.add(1, { ...opts.labels, result: 'completed' });
+      } catch (ex) {
+        this.counter.add(1, { ...opts.labels, result: 'failed' });
+        throw ex;
+      } finally {
+        const delta = process.hrtime(startTime);
+        const endTime = delta[0] + delta[1] / 1e9;
+        this.duration.record(endTime, { ...opts.labels });
```

Comment:
You want to label completion with `result: 'completed'` / `'failed'` too. Things that fail typically have wildly different timings than those that don't.